extension-style/src/main/java/com/mapbox/maps/extension/style/layers/Layer.kt

```diff
@@ -100,6 +100,11 @@ abstract class Layer : StyleContract.StyleLayerExtension {
    */
   fun bindTo(delegate: StyleInterface) {
     bindTo(delegate, null)
+    (this.getCachedLayerProperties().contents as? Value?)?.let { value ->
+      (value.contents as HashMap<String, Value>).forEach {
+        delegate.addStyleLayer(it.value, null)
+      }
+    }
```

Comment:
shouldn't these logic be added to the `fun bindTo(delegate: StyleInterface, position: LayerPosition?)` instead of here?